### ISSUE TITLE
DeprecationWarning in future/type/newstr.py importing from "collections" instead of "collections.abc"

### ISSUE THREAD

**Issue by ThreePinkApples** [PythonCharmers/python-future#469]
https://github.com/PythonCharmers/python-future/blob/master/src/future/types/newstr.py#L43

```
lib/python3.7/site-packages/future/types/newstr.py:43: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Iterable
```

**Comment by Jonahss** [comment]
Would changing the imports for collections this way be a good fix?
```
try:
    from collections.abc import Iterable  # noqa
except ImportError:
    from collections import Iterable  # noqa
```

**Comment by ThreePinkApples** [comment]
collections.abc was introduced in Python 3.3, so imo it is better to make it explicit with if/else. I'm assuming supporting Python 3.0-3.2 isn't necessary, so then you can do this:
```
if sys.version_info.major > 2:
    from collections.abc import Iterable
else:
    from collections import Iterable
```

**Comment by jmadler** [comment]
I believe this is resolved in 0.18.0 -- please test and let me know if you have any issues.
